<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Title</title>
</head>
<body>
<div>
    <h2>Report about the Long Non-coding RNA (lncRNA) HOTAIR</h2>
    <p>Based on lncRNA-gene co-expression, this report provides predictions about the biological functions of HOTAIR, displays the average expression of HOTAIR across tissues and cell-lines, and lists predictions of small molecules that may specifically up- or down-regulate the expression of HOTAIR.</p>
    <h3>Top correlated genes with HOTAIR</h3>
    <p>Using the loaded gene-lncRNA correlation matrix, we report the genes that mostly correlate with HOTAIR.</p>
    <table>
        <tr>
            <th>Gene</th>
            <th>Pearson's Correlation Coefficient</th>
        </tr>
        <tr>
            <td>HOXC11</td>
            <td>0.911</td>
        </tr>
        <tr>
            <td>ENSG00000277994</td>
            <td>0.838</td>
        </tr>
        <tr>
            <td>HOXC-AS3</td>
            <td>0.829</td>
        </tr>
        <tr>
            <td>HOXC10</td>
            <td>0.819</td>
        </tr>
        <tr>
            <td>ENSG00000275589</td>
            <td>0.800</td>
        </tr>
    </table>
    <p style="font-style: italic;">Table 1. Top 20 genes that mostly correlate with HOTAIR ranked by Pearson’s correlation coefficients.</p>
    <p>Download Table 1: <a href="">gene_correlations/HOTAIR_correlated_genes.csv</a></p>
    <h3>Enrichment analysis applied to the top 200 most correlated genes with HOTAIR</h3>
    <p>The top 200 most correlated genes with HOTAIR were submitted to Enrichr [2-4] for enrichment analysis. NOTE: Only genes with official Entrez gene symbols are submitted to Enrichr. Ensembl IDs that do not map to an official gene symbol were dropped.</p>
    <p>Access the enrichment analysis results for HOTAIR 200 most correlated genes here: <a href="">https://maayanlab.cloud/Enrichr/enrich?dataset=564bc230f69acf57b31ab657c7602179</a></p>
    <h3>Top correlated lncRNAs with HOTAIR</h3>
    <p>Below we list the top lncRNAs, out of all 5,050 lncRNAs within our database, that mostly correlate with HOTAIR based on their Pearson’s correlation coefficients.</p>
    <table>
        <tr>
            <th>Gene</th>
            <th>Pearson's Correlation Coefficient</th>
        </tr>
        <tr>
            <td>HOXC-AS3</td>
            <td>0.829</td>
        </tr>
        <tr>
            <td>HOXC13-AS</td>
            <td>0.759</td>
        </tr>
        <tr>
            <td>HOXC-AS1</td>
            <td>0.740</td>
        </tr>
        <tr>
            <td>HOXC-AS2</td>
            <td>0.730</td>
        </tr>
        <tr>
            <td>HOXA11-AS</td>
            <td>0.618</td>
        </tr>
    </table>
    <p>Table 2. Top 20 lncRNAs that mostly correlate with HOTAIR ranked by Pearson’s correlation coefficients.</p>
    <p>Download Table 2: <a href="">gene_correlations/HOTAIR_correlated_lncRNAs.csv</a></p>
    <h3>Predicted Biological Functions of HOTAIR</h3>
    <p>For each Enrichr library, the mean Pearson’s correlation coefficients are calculated between each gene set and HOTAIR. Terms with a high mean Pearson’s correlation coefficients are prioritized. These terms are predicted to be associated with HOTAIR.</p>
    <img src="" />
    <p style="font-style: italic;">Figure 1. Predicted MGI Mammalian Phenotypes and GO Biological Processes for the lncRNA HOTAIR. Terms are ranked by the mean Pearson correlation between HOTAIR and the associated gene sets.</p>
    <p>Download predictions: <a href="">predicted_functions/MGI Mammalian Phenotype Level 4 2021_HOTAIR.csv</a></p>
    <p>Download predictions: <a href="">predicted_functions/GO Biological Process 2021_HOTAIR.csv</a></p>
    <img src="" />
    <p style="font-style: italic;">Figure 2. Predicted KEGG pathways and DisGeNET disease terms for the lncRNA HOTAIR. Terms are ranked by the mean Pearson correlation between HOTAIR and the associated gene sets.</p>
    <p>Download predictions: <a href="">predicted_functions/KEGG 2021 Human_HOTAIR.csv</a></p>
    <p>Download predictions: <a href="">predicted_functions/DisGeNET_HOTAIR.csv</a></p>

    <h3>Tissue and Cell Line Expression Levels of HOTAIR</h3>
    <p style="font-weight: 600">This part of the report provides the Z-score (Normalized Median Expression) for the lncRNA in various tissues and cell lines.
    <p>Samples from Recount3 [1] were automatically labelled by their tissue type or cell line of origin. Tissue and cell line samples were log2 transformed and quantile normalized separately. For HOTAIR, the median expression was then calculated for each tissue type and cell line. Tissues and cell lines with less than 20 samples were removed. Z-scoring was then applied along the lncRNA axis to compare expression levels across all tissues and cell lines.</p>

    <h3>References</h3>
    <p>[1] Wilks C, Zheng SC, Chen FY, Charles R, Solomon B, Ling JP, Imada EL, Zhang D, Joseph L, Leek JT: recount3: summaries and queries for large-scale RNA-seq expression and splicing. bioRxiv 2021:2021.2005.2021.445138.</p>
    <p>[2] Xie Z, Bailey A, Kuleshov MV, Clarke DJB, Evangelista JE, Jenkins SL, Lachmann A, Wojciechowicz ML, Kropiwnicki E, Jagodnik KM: Gene Set Knowledge Discovery with Enrichr. Current Protocols 2021, 1(3):e90.</p>
    <p>[3] Chen EY, Tan CM, Kou Y, Duan Q, Wang Z, Meirelles GV, Clark NR, Ma’ayan A: Enrichr: interactive and collaborative HTML5 gene list enrichment analysis tool. BMC Bioinformatics 2013, 14(1):128.</p>
    <p>[4] Kuleshov MV, Jones MR, Rouillard AD, Fernandez NF, Duan Q, Wang Z, Koplev S, Jenkins SL, Jagodnik KM, Lachmann A: Enrichr: a comprehensive gene set enrichment analysis web server 2016 update. Nucleic Acids Research 2016, 44(W1):W90-W97.</p>
    <p>[5] McInnes L, Healy J, Melville J: Umap: Uniform manifold approximation and projection for dimension reduction. arXiv preprint arXiv:180203426 2018.</p>
    <p>[6] Evangelista et al. SigCom LINCS: Data and Metadata Search Engine for Gene Expression Signatures. 2021. In preparation.</p>
</div>
</body>
</html>
