{% extends "base.html" %}

{% block body %}
<div class="container mt-5">
    <h4>Report about the Long Non-coding RNA (lncRNA) {{ appyter.gene }} <a class="ml-2 p-1" style="font-size: 1rem; font-weight: 400; background: aliceblue; border-radius: 4px; border: 1px solid #bddffd;" target="_blank" rel="noopener noreferrer" href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}">Open in <img src="{{ url_for('static', filename='appyters_logo.svg') }}" class="img-fluid" style="width: 10%" alt="Appyters"></a></h4>
    <p>Based on lncRNA-gene co-expression, this report provides predictions on the biological functions of {{
        appyter.gene }},
        displays the median expression of {{ appyter.gene }} across tissues and cell-lines, and predicts small molecules
        that may
        specifically up- or down-regulate the expression of {{ appyter.gene }}.</p>

    <div class="row">
        <div class="col-sm">
            <h5 class="mt-4">Top genes correlated with {{ appyter.gene }}</h5>
            <p style="height: 5rem;">Using the loaded lncRNA-gene correlation matrix, we report the genes that are most correlated with {{
                appyter.gene }}.</p>
            <div class="table-responsive-sm">
                <table id="table1-{{ appyter.id }}" class="compact hover order-column row-border"
                       style="width:100%"></table>
            </div>
            <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a
                    href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/gene_correlations/{{ appyter.gene }}_correlated_genes.csv"><i class="fas fa-file-download"></i></a> Table 1. The Top 20 genes correlated with {{ appyter.gene }} ranked by
                Pearson’s correlation coefficients.</p>
        </div>
        <div class="col-sm">
            <h5 class="mt-4">Top lncRNAs correlated with {{ appyter.gene }}</h5>
            <p style="height: 5rem;">Below we list the top 20 lncRNAs, out of all 15,862 lncRNAs within our database, that correlate most with
                {{ appyter.gene }} based on their Pearson correlation coefficients.</p>
            <div class="table-responsive-sm">
                <table id="table2-{{ appyter.id }}" class="compact hover order-column row-border"
                       style="width:100%"></table>
            </div>
            <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a
                    href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/gene_correlations/{{ appyter.gene }}_correlated_lncRNAs.csv"><i class="fas fa-file-download"></i></a> Table 2. Top 20 lncRNAs that correlate most with {{ appyter.gene }} ranked by
                Pearson correlation coefficients.</p>
        </div>
    </div>

    <h5 class="mt-4">Interactive network visualization of the top 100 genes correlated with {{ appyter.gene }}</h5>
    <p>Interactive network visualization of the top 100 genes correlated with {{ appyter.gene }}. Each node represents a
        gene and is colored by chromosome location, except for the bright red node which represents the lncRNA {{
        appyter.gene }}. The thickness of the edges corresponds to Pearson correlation coefficients. Clicking on a gene
        node will highlight its corresponding edges in orange. Hovering over a node will display the gene name and
        chromosome location.</p>
    <p>Network Methods: All pairwise correlations between the top 100 genes correlated with {{ appyter.gene }} are
        extracted. The 3 edges with the highest correlation per gene node are used to initialize the network. Edges with
        weights < 0.3 are dropped. To further prune the network, the edge with the lowest weight for each hub node is
        dropped. At the start, a hub node is defined as a node with > 10 edges. The pruning process is repeated until
        the network has an average of < 3 edges per node. The top 5 edges for {{ appyter.gene }} are shown regardless of
        their weights.</p>
    <p class="mb-2" style="font-size: 0.8rem; font-style: italic;">Figure 1. Interactive network visualization of the top 100 genes correlated with {{
        appyter.gene }}.</p>

    <p class="mb-2" style="font-size: 0.8rem">Download Figure 1: <a
            href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/coexpression_network/{{ appyter.gene }}_network.html">coexpression_network/{{
        appyter.gene }}_network.html</a></p>
    <p class="mb-2" style="font-size: 0.8rem">Download Node metadata: <a
            href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/coexpression_network/{{ appyter.gene }}_network_node_metadata.csv">coexpression_network/{{
        appyter.gene }}_network_node_metadata.csv</a></p>
    <p class="mb-2" style="font-size: 0.8rem">Download Edge metadata: <a
            href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/coexpression_network/{{ appyter.gene }}_network_edge_metadata.csv">coexpression_network/{{
        appyter.gene }}_network_edge_metadata.csv</a></p>

    <h5 class="mt-4">Predicted biological functions of {{ appyter.gene }}</h5>
    <p>For each Enrichr library, we compute the mean Pearson correlation coefficients for each gene set by averaging the
        Pearson correlation coefficients between each gene in the gene set and {{ appyter.gene }}. Terms with high mean
        Pearson correlation coefficients are prioritized. These terms are predicted to be associated with {{
        appyter.gene }}.</p>

    <div class="row">
        <div class="col-sm">
            <img src="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/predicted_functions/{{ appyter.gene }}_biological_function_predictions_figure2.png"
                 class="img-fluid p-3 bg-light"
                 alt="Figure 2. Predicted MGI Mammalian Phenotypes and GO Biological Processes for the lncRNA {{ appyter.gene }}. Terms are ranked by averaging the mean Pearson correlation coefficients between each gene in a gene set and {{ appyter.gene }}.">
            <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">Figure 2. Predicted MGI Mammalian Phenotypes and GO
                Biological Processes for
                the
                lncRNA {{ appyter.gene }}. Terms are ranked by averaging the mean Pearson correlation coefficients
                between each
                gene in a gene set and {{ appyter.gene }}.</p>
            <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a
                    href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/predicted_functions/MGI Mammalian Phenotype Level 4 2021_{{ appyter.gene }}.csv">predicted_functions/MGI
                Mammalian Phenotype Level 4 2021_{{ appyter.gene }}.csv</a></p>
            <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a
                    href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/predicted_functions/GO Biological Process 2021_{{ appyter.gene }}.csv">predicted_functions/GO
                Biological Process 2021_{{ appyter.gene }}.csv</a></p>
        </div>
        <div class="col-sm">
            <img src="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/predicted_functions/{{ appyter.gene }}_biological_function_predictions_figure3.png"
                 class="img-fluid p-3 bg-light"
                 alt="Figure 3. Predicted KEGG pathways and DisGeNET disease terms for the lncRNA {{ appyter.gene }}. Terms are ranked by averaging the mean Pearson correlation coefficients between each gene in a gene set and {{ appyter.gene }}.">
            <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">Figure 3. Predicted KEGG pathways and DisGeNET
                disease terms for the lncRNA {{
                appyter.gene }}. Terms are ranked by averaging the mean Pearson correlation coefficients between each
                gene in a
                gene set and {{ appyter.gene }}.</p>
            <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a
                    href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/predicted_functions/KEGG 2021 Human_{{ appyter.gene }}.csv">predicted_functions/KEGG
                2021 Human_{{ appyter.gene }}.csv</a></p>
            <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a
                    href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/predicted_functions/DisGeNET_{{ appyter.gene }}.csv">predicted_functions/DisGeNET_{{
                appyter.gene }}.csv</a></p>
        </div>
    </div>

    <h5 class="mt-4">Expression of {{ appyter.gene }} across tissues and cell lines</h5>
    <p>This part of the report provides the Z-score (Normalized Median Expression) for the
        lncRNA {{ appyter.gene }} in various tissues and cell lines. Samples from Recount3 [1] were automatically
        labelled by tissue type or cell line of origin. Tissue and cell line
        samples were log2 transformed and quantile normalized separately. Tissues and cell lines with less than 20
        samples were removed, and z-scores were computed along the lncRNA axis to compare expression levels across all
        tissues and cell lines. The median expression of {{ appyter.gene }} was then calculated for each tissue type and
        cell line.</p>

    <div class="row">
        <div class="col-sm">
            <img src="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/tissue_and_cell_line_expression/{{ appyter.gene }}_zscore_tissue_expression.png"
                 class="img-fluid p-3 bg-light"
                 alt="Figure 4. Z-score (median expression) for the lncRNA {{ appyter.gene }} in various tissue types.">
            <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">Figure 4. Z-score (median expression) for the lncRNA
                {{ appyter.gene }} in
                various
                tissue types.</p>
            <p class="mb-2" style="font-size: 0.8rem">Download table with z-score (median expression) values for {{ appyter.gene }}
                in various tissue types: <a
                        href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/tissue_and_cell_line_expression/{{ appyter.gene }}_tissue_zscore.csv">tissue_and_cell_line_expression/{{
                    appyter.gene }}_tissue_zscore.csv</a></p>
        </div>
        <div class="col-sm">
            <img src="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/tissue_and_cell_line_expression/{{ appyter.gene }}_zscore_cell_line_expression.png"
                 class="img-fluid p-5 bg-light"
                 alt="Figure 5. Z-score (median expression) for the lncRNA {{ appyter.gene }} in the top 30 cell lines.">
            <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">Figure 5. Z-score (median expression) for the lncRNA
                {{ appyter.gene }} in the
                top 30
                cell lines.</p>
            <p class="mb-2" style="font-size: 0.8rem">Download table with z-score (median expression) values for {{ appyter.gene }}
                in various cell lines: <a
                        href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/tissue_and_cell_line_expression/{{ appyter.gene }}_cell_line_zscore.csv">tissue_and_cell_line_expression/{{
                    appyter.gene }}_cell_line_zscore.csv</a></p>
        </div>
    </div>
    <h5 class="mt-4">Visualizing lncRNAs based on their gene expression similarity across tissues and cell lines</h5>
    <p>We applied UMAP [5] to visualize lncRNA expression across 3,000 randomly selected samples (with tissue type
        and cell line labels) from Recount3[1]. Samples were first log2 transformed and quantile normalized along
        the gene axis, then UMAP was applied to the lncRNA expression data with samples as features. Each data point
        represents a single lncRNA (n=15,862). The black arrow is pointing to the location of {{ appyter.gene
        }}.</p>
    <div class="row">
        <div class="col-sm">
            <img src="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/umap/tissues/figures/static/{{ appyter.gene }}_{{ appyter.data.fig6_tissue }}_rank1.png"
                 class="img-fluid p-3 bg-light"
                 alt="Figure 6. UMAP was applied to 3,000 randomly selected samples (with tissue type labels) from Recount3. Each data point represents a lncRNA (n=15,862) and are colored by z-score (median expression) in {{ appyter.data.fig6_tissue }}.">
            <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">Figure 6. UMAP was applied to 3,000 randomly
                selected samples (with tissue type labels) from Recount3.
                Each data
                point represents a lncRNA (n=15,862) and are colored by z-score (median expression) in {{
                appyter.data.fig6_tissue }}.</p>
        </div>
        <div class="col-sm">
            <img src="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/umap/cell_lines/figures/static/{{ appyter.gene }}_{{ appyter.data.fig7_cell_line }}_rank1.png"
                 class="img-fluid p-3 bg-light"
                 alt="Figure 7. UMAP was applied to 3,000 randomly selected samples (with cell line labels) from Recount3. Each data point represents a lncRNA (n=15,862) and are colored by z-score (median expression) in {{ appyter.data.fig6_cell_line }}.">
            <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">Figure 7. UMAP was applied to 3,000 randomly
                selected samples (with cell line labels) from Recount3. Each
                data
                point represents a lncRNA (n=15,862) and are colored by z-score (median expression) in {{
                appyter.data.fig7_cell_line }}.</p>
        </div>
    </div>

    <h5 class="mt-4">L1000 small molecules predicted to modulate {{ appyter.gene }}</h5>
    <p>~1.4 million Level 5 L1000 chemical perturbation gene expression signatures were downloaded from SigCom LINCS
        (https://maayanlab.cloud/sigcom-lincs) [6]. For each unique signature and lncRNA pair, a mean Pearson
        correlation coefficient was computed by taking the average Pearson coefficient between the lncRNA and all genes
        in the signature. All 15,862 lncRNAs were then ranked by mean Pearson correlation coefficient, and the top 1,000
        lncRNAs with the highest coefficients were retained for each signature. The top 500 lncRNA-L1000 signature
        associations are reported here for {{ appyter.gene }}, separated by direction. If {{ appyter.gene }} is highly
        correlated with the
        up-regulated genes for a specific small molecule, then this small molecule is predicted to up-regulate
        {{ appyter.gene }}.</p>
    <h5 class="mt-4">L1000 small molecules predicted to up-Regulate {{ appyter.gene }}</h5>
    <p>The prioritized small molecules below are predicted to specifically up-regulate {{ appyter.gene }}.</p>

    <div class="table-responsive-sm">
        <table id="table3-{{ appyter.id }}" class="compact hover order-column row-border"
               style="width:100%"></table>
    </div>

    <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a
            href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/l1000_sm_predictions/{{ appyter.gene }}_l1000_sm_predictions_up.csv"><i class="fas fa-file-download"></i></a> Table 3. L1000 small molecules predicted to up-regulate the lncRNA {{ appyter.gene
        }}.</p>

    <h5 class="mt-4">L1000 small molecules predicted to down-regulate {{ appyter.gene }}</h5>
    <p>The prioritized small molecules below are predicted to specifically down-regulate {{ appyter.gene }}.</p>

    <div class="table-responsive-sm">
        <table id="table4-{{ appyter.id }}" class="compact hover order-column row-border"
               style="width:100%"></table>
    </div>
    <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a
                href="https://appyters.maayanlab.cloud/lncRNA_Appyter/{{ appyter.id}}/l1000_sm_predictions/{{ appyter.gene }}_l1000_sm_predictions_down.csv"><i class="fas fa-file-download"></i></a> Table 4. L1000 small molecules predicted to down-regulate the lncRNA {{ appyter.gene
        }}.</p>


    <h5 class="mt-4">References</h5>
    <p class="mb-1" style="font-size: 0.8rem;">[1] Wilks C, Zheng SC, Chen FY, Charles R, Solomon B, Ling JP, Imada EL, Zhang D, Joseph L, Leek JT: Recount3:
        summaries and queries for large-scale RNA-seq expression and splicing. bioRxiv 2021:2021.2005.2021.445138.</p>
    <p class="mb-1" style="font-size: 0.8rem;">[2] Xie Z, Bailey A, Kuleshov MV, Clarke DJB, Evangelista JE, Jenkins SL, Lachmann A, Wojciechowicz ML,
        Kropiwnicki E, Jagodnik KM: Gene Set Knowledge Discovery with Enrichr. Current Protocols 2021, 1(3):e90.</p>
    <p class="mb-1" style="font-size: 0.8rem;">[3] Chen EY, Tan CM, Kou Y, Duan Q, Wang Z, Meirelles GV, Clark NR, Ma’ayan A: Enrichr: interactive and
        collaborative HTML5 gene list enrichment analysis tool. BMC Bioinformatics 2013, 14(1):128.</p>
    <p class="mb-1" style="font-size: 0.8rem;">[4] Kuleshov MV, Jones MR, Rouillard AD, Fernandez NF, Duan Q, Wang Z, Koplev S, Jenkins SL, Jagodnik KM,
        Lachmann A: Enrichr: a comprehensive gene set enrichment analysis web server 2016 update. Nucleic Acids Research
        2016, 44(W1):W90-W97.</p>
    <p class="mb-1" style="font-size: 0.8rem;">[5] McInnes L, Healy J, Melville J: Umap: Uniform manifold approximation and projection for dimension reduction.
        arXiv preprint arXiv:180203426 2018.</p>
    <p class="mb-1" style="font-size: 0.8rem;">[6] Evangelista et al. SigCom LINCS: Data and Metadata Search Engine for Gene Expression Signatures. 2021. In
        preparation.</p>

</div>
{% endblock %}

{% block script %}
<script>
    let appyter_id = "{{ appyter.id|safe }}";
    let gene = "{{ appyter.gene|safe }}"
</script>
<script type="text/javascript" src="{{ url_for('static', filename='appyter.js') }}"></script>
{% endblock %}