{% extends "base.html" %}

{% block og %}
<meta property="og:url" content="https://{{ base_path }}/lncHUB2"/>
<meta property="og:description"
      content="Functional predictions of human long non-coding RNAs based on lncRNA-gene co-expression correlations."/>
{% endblock %}
{% block head %}
<title>lncHUB2: Predicted biological functions for long non-coding RNAs based on co-expression</title>
{% endblock %}

{% block body %}
<div class="container">
    <!--  Header  -->
    <div class="row mt-5 justify-content-center">
        <div class="col-1">
            <img class="img-fluid float-end" style="width: 65px; min-height: 32px; min-width: 40px;" src="{{ url_for('static', filename='logo.png') }}"
                 alt="lncRNA logo">
        </div>
        <div class="col-2 ps-0">
            <h2 class="display-5">lncHUB2</h2>
        </div>
    </div>
    <div class="row justify-content-center">
        <div class="col-7">
            <p style="font-weight: 200; font-size: 1.6rem; line-height: 1.3;" class="text-center">Functional predictions
                of human long
                non-coding RNAs based on lncRNA-gene co-expression correlations.</p>
        </div>
    </div>
    <!--  Header  -->
    <!-- Search -->
    <div class="row justify-content-center my-5">
        <div class="col-8">
            <p>A long non-coding RNA (lncRNA) is a transcript with more than 200 nucleotides that is not translated into
                protein. Based on gene-gene co-expression correlations created from Recount3's processed RNA-seq
                samples, we present 15,862 landing pages for long non-coding RNAs that include median expression across
                tissues and cell lines, predicted biological functions, pathway membership, and predicted small
                molecules that may regulate their expression.</p>
            <div class="input-group mt-4 mb-2">
                <input id="search" type="text" autocapitalize="on" maxlength="50" class="form-control"
                       aria-label="lncRNA"
                       aria-describedby="search-button"/>
                <button class="btn btn-outline-secondary" type="button" id="search-button"
                        onclick="search($('#search').val())">Search
                </button>
            </div>
            <p class="small text-muted">For example <a href="#" onclick="example('HOTAIR');">HOTAIR</a>, <a href="#"
                                                                                                            onclick="example('MALAT1');">MALAT1</a>,
                <a href="#" onclick="example('ENSG00000245532');">ENSG00000245532</a></p>
        </div>
    </div>
    <!-- Search -->
    <!-- Results -->
    <!-- Appyter Loader -->

    <div id="appyter-loader" class="row justify-content-center my-3" style="display: none;">
        <p>Loading...</p>
    </div>
    <!--  Not lncRNA  -->
    <div id="not-lncRNA" class="row justify-content-center my-3" style="display: none;">
        <p id="not-lncRNA-msg" style="font-style: italic;"></p>
    </div>
    <!-- Appyter Loader -->
    <div id="results" class="row justify-content-center my-3">
        <div class="col-2">
            <nav id="navbar-toc"
                 class="navbar navbar-light bg-light flex-column align-items-stretch p-3 sticky-top" style="display: none;">
                <a class="navbar-brand" href="#">Report about the Long Non-coding RNA (lncRNA) <span class="gene-name"></span></a>
                <nav class="nav nav-pills flex-column">
                    <a class="nav-link" href="#item-1">Top genes and lncRNAs correlated with <span class="gene-name"></span></a>
                    <a class="nav-link" href="#item-2">Interactive network visualization of the top 100 genes correlated with <span
                            class="gene-name"></span></a>
                    <a class="nav-link" href="#item-3">Predicted biological functions of <span class="gene-name"></span></a>
                    <nav class="nav nav-pills flex-column">
                        <a class="nav-link" href="#item-3-1">Enrichment analysis applied to the top 200 genes most correlated with <span
                        class="gene-name"></span></a>
                        </nav>
                    <a class="nav-link" href="#item-4">Expression of <span class="gene-name"></span> across tissues and cell lines</a>
                    <a class="nav-link" href="#item-5">Visualizing lncRNAs based on their gene expression similarity across tissues and cell
                    lines</a>
                    <a class="nav-link" href="#item-6">L1000 small molecules predicted to modulate <span class="gene-name"></span></a>
                    <nav class="nav nav-pills flex-column">
                        <a class="nav-link" href="#item-6-1">L1000 small molecules predicted to up-regulate <span class="gene-name"></span></a>
                        <a class="nav-link" href="#item-6-2">L1000 small molecules predicted to down-regulate <span class="gene-name"></span></a>
                    </nav>
                    <a class="nav-link" href="#item-7">References</a>
                </nav>
            </nav>
        </div>
        <p id="results__not-in-db" style="font-style: italic; display: none;"></p>
        <div id="results__appyter-card" class="col-9 card m-2" style="display: none; position: relative">
            <div class="card-body" style="position: relative" data-bs-spy="scroll" data-bs-target="#navbar-toc">
                <h4 class="card-title">Report about the Long Non-coding RNA (lncRNA) <span class="gene-name"></span> <a
                        class="p-1"
                        id="appyter-url"
                        style="font-size: 1rem; font-weight: 400; background: aliceblue; border-radius: 4px; border: 1px solid #bddffd;"
                        target="_blank"
                        rel="noopener noreferrer"
                        href=""><span id="appyter-action"></span><img
                        src="{{ url_for('static', filename='appyters_logo.svg') }}" class="img-fluid"
                        style="width: 120px"
                        alt="Appyters"></a></h4>
                <p>Based on lncRNA-gene co-expression, this report provides predictions on the biological functions of
                    <span
                            class="gene-name"></span>,
                    displays the median expression of <span class="gene-name"></span> across tissues and cell-lines, and
                    predicts
                    small molecules
                    that may
                    specifically up- or down-regulate the expression of <span class="gene-name"></span>.</p>

                <div class="row justify-content-center">
                    <div class="col-10 col-sm-6">
                        <img id="struct-img" src="" class="img-fluid" alt="" style="border: 0.25px solid darkgray; border-radius: 4px;">
                        <p id="struct-desc" style="font-size: 0.8rem; font-style: italic;"><a
                                id="struct-img-down"
                                href=""><i
                                class="fas fa-file-download"></i></a> Secondary structure of <span
                                class="gene-name"></span> predicted with RNAfold [1].</p>
                    </div>
                </div>

                <div class="row">
                    <div class="col-sm">
                        <h5 id="item-1" class="mt-4">Top genes correlated with <span class="gene-name"></span></h5>
                        <p style="height: 5rem;">Using the loaded lncRNA-gene correlation matrix, we report the genes
                            that are most
                            correlated with <span class="gene-name"></span>.</p>
                        <div class="table-responsive-sm">
                            <p id="table-blank1" style="display: none; font-style: italic;">There are no lncRNAs
                                correlated with <span class="gene-name"></span>.</p>
                            <table id="table1" class="compact hover order-column row-border"
                                   style="width:100%"></table>
                        </div>
                        <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a
                                id="tab1-down"
                                href=""><i
                                class="fas fa-file-download"></i></a> Table 1. The Top 20 genes correlated with <span
                                class="gene-name"></span> ranked by
                            Pearsonâ€™s correlation coefficients.</p>
                    </div>
                    <div class="col-sm">
                        <h5 class="mt-4">Top lncRNAs correlated with <span class="gene-name"></span></h5>
                        <p style="height: 5rem;">Below we list the top 20 lncRNAs, out of all 15,862 lncRNAs within our
                            database,
                            that correlate most with
                            <span class="gene-name"></span> based on their Pearson correlation coefficients.</p>
                        <div class="table-responsive-sm">
                            <p id="table2-blank" style="display: none; font-style: italic;">There are no lncRNAs
                                correlated with <span class="gene-name"></span>.</p>
                            <table id="table2" class="compact hover order-column row-border"
                                   style="width:100%"></table>
                        </div>
                        <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a
                                id="tab2-down"
                                href=""><i
                                class="fas fa-file-download"></i></a> Table 2. Top 20 lncRNAs that correlate most with
                            <span
                                    class="gene-name"></span> ranked by
                            Pearson correlation coefficients.</p>
                    </div>
                </div>

                <h5 id="item-2" class="mt-4">Interactive network visualization of the top 100 genes correlated with <span
                        class="gene-name"></span></h5>
                <p>Interactive network visualization of the top 100 genes correlated with <span
                        class="gene-name"></span>. Each node
                    represents a
                    gene and is colored by chromosome location, except for the bright red node which represents the
                    lncRNA <span
                            class="gene-name"></span>. The thickness of the edges corresponds to Pearson correlation
                    coefficients.
                    Clicking on a gene
                    node will highlight its corresponding edges in orange. Hovering over a node will display the gene
                    name and
                    chromosome location.</p>
                <p>Network Methods: All pairwise correlations between the top 100 genes correlated with <span
                        class="gene-name"></span> are
                    extracted. The 3 edges with the highest correlation per gene node are used to initialize the
                    network. Edges with
                    weights < 0.3 are dropped. To further prune the network, the edge with the lowest weight for each
                    hub node is
                    dropped. At the start, a hub node is defined as a node with > 10 edges. The pruning process is
                    repeated until
                    the network has an average of < 3 edges per node. The top 5 edges for <span
                            class="gene-name"></span> are shown
                    regardless of
                    their weights.</p>

                <p class="mb-2" style="font-size: 0.8rem">Download interactive network
                    visualization of the
                    top 100 genes correlated with <span class="gene-name"></span>: <a
                            id="appyter-fig1-net"
                            href="">coexpression_network/<span
                            class="gene-name"></span>_network.html</a></p>
                <p class="mb-2" style="font-size: 0.8rem">Download Node metadata: <a
                        id="appyter-fig1-node-meta"
                        href="">coexpression_network/<span
                        class="gene-name"></span>_network_node_metadata.csv</a></p>
                <p class="mb-2" style="font-size: 0.8rem">Download Edge metadata: <a
                        id="appyter-fig1-edge-meta"
                        href="">coexpression_network/<span
                        class="gene-name"></span>_network_edge_metadata.csv</a></p>

                <h5 id="item-3" class="mt-4">Enrichment analysis applied to the top 200 genes most correlated with <span
                        class="gene-name"></span></h5>

                <p>The top 200 genes most correlated with <span class="gene-name"></span> are submitted to Enrichr [3-5]
                    for
                    enrichment analysis. NOTE: Only genes with official Entrez gene symbols are submitted to Enrichr.
                    Ensembl IDs
                    that do not map to an official gene symbol were dropped.</p>
                <p><a id="appyter-enrichr-url" href="" target="_blank">Access the enrichment analysis results</a> at
                    Enrichr.</p>


                <h5 id="item-3-1" class="mt-4">Predicted biological functions of <span class="gene-name"></span></h5>
                <p>For each Enrichr library, we compute the mean Pearson correlation coefficients for each gene set by
                    averaging the
                    Pearson correlation coefficients between each gene in the gene set and <span
                            class="gene-name"></span>. Terms
                    with high mean
                    Pearson correlation coefficients are prioritized. These terms are predicted to be associated with
                    <span
                            class="gene-name"></span>.</p>

                <div class="row">
                    <div class="col-sm">
                        <img id="fig2-img" src="" class="img-fluid p-3 bg-light" alt="">
                        <p id="fig2-desc" style="font-size: 0.8rem; font-style: italic;"
                           class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig2-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 2. Predicted MGI Mammalian Phenotypes and GO Biological Processes for the lncRNA
                            <span class="gene-name"></span>. Terms are ranked by averaging the Pearson correlation
                            coefficients between each gene in a gene set and <span class="gene-name"></span>.
                        </p>
                        <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a id="fig2-down1"
                                                                                           href="">predicted_functions/MGI
                            Mammalian Phenotype Level 4 2021_<span class="gene-name"></span>.csv</a></p>
                        <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a id="fig2-down2"
                                                                                           href="">predicted_functions/GO
                            Biological Process 2021_<span class="gene-name"></span>.csv</a></p>
                        <div class="modal fade" id="fig2-mod" tabindex="-1" aria-labelledby="fig2-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig2-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig2-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>

                    <div class="col-sm">
                        <img id="fig3-img" src=""
                             class="img-fluid p-3 bg-light"
                             alt="">
                        <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig3-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 3. Predicted KEGG pathways and DisGeNET disease terms for the lncRNA <span
                                class="gene-name"></span>. Terms are ranked by averaging the Pearson correlation
                            coefficients between each gene in a gene set and <span class="gene-name"></span>.
                        </p>
                        <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a id="fig3-down1"
                                                                                           href="">predicted_functions/KEGG
                            2021 Human_<span class="gene-name"></span>.csv</a></p>
                        <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a id="fig3-down2"
                                                                                           href="">predicted_functions/DisGeNET_<span
                                class="gene-name"></span>.csv</a></p>
                        <div class="modal fade" id="fig3-mod" tabindex="-1" aria-labelledby="fig3-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig3-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig3-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>

                <div class="row">
                    <div class="col col-sm-6">
                        <img id="fig8-img" src="" class="img-fluid p-3 bg-light" alt="">
                        <p id="fig8-desc" style="font-size: 0.8rem; font-style: italic;"
                           class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig8-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 4. Predicted transcription factors from ChEA 2016 and ENCODE ChIP-seq 2015 libraries
                            for the lncRNA <span class="gene-name"></span>. Terms are ranked by averaging the Pearson
                            correlation coefficients between each gene in a gene set and <span class="gene-name"></span>.
                        </p>
                        <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a id="fig8-down1"
                                                                                           href="">predicted_functions/ChEA_<span
                                class="gene-name"></span>.csv</a></p>
                        <p class="mb-2" style="font-size: 0.8rem">Download predictions: <a id="fig8-down2"
                                                                                           href="">predicted_functions/ENCODE_<span
                                class="gene-name"></span>.csv</a></p>
                        <div class="modal fade" id="fig8-mod" tabindex="-1" aria-labelledby="fig8-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig8-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig8-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>

                <h5 id="item-4" class="mt-4">Expression of <span class="gene-name"></span> across tissues and cell lines</h5>
                <p>This part of the report provides the Z-score (Normalized Median Expression) for the lncRNA <span
                        class="gene-name"></span> in various tissues and cell lines. Samples from Recount3 [2] were
                    automatically labelled by tissue type or cell line of origin. Tissue and cell line samples were log2
                    transformed and quantile normalized separately. Tissues and cell lines with less than 20 samples
                    were removed and the median expression of <span class="gene-name"></span> was then calculated for
                    each tissue type and cell line. Z-scores were computed along the lncRNA axis to compare expression
                    levels across all tissues and cell lines.</p>

                <div class="row">
                    <div class="col-sm">
                        <img id="fig4-img" src=""
                             class="img-fluid p-3 bg-light"
                             alt="">
                        <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig4-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 5.
                            Z-score (median
                            expression) for the lncRNA
                            <span class="gene-name"></span> in
                            various
                            tissue types.
                        </p>
                        <p class="mb-2" style="font-size: 0.8rem">Download table with z-score (median expression) values
                            for <span
                                    class="gene-name"></span>
                            in various tissue types: <a id="fig4-down" href="">tissue_and_cell_line_expression/<span
                                    class="gene-name"></span>_tissue_zscore.csv</a></p>
                        <div class="modal fade" id="fig4-mod" tabindex="-1" aria-labelledby="fig4-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig4-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig4-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="col-sm">
                        <img id="fig5-img" src=""
                             class="img-fluid p-5 bg-light"
                             alt="">
                        <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig5-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 6.
                            Z-score (median
                            expression) for the lncRNA
                            <span class="gene-name"></span> in the
                            top 30
                            cell lines.
                        </p>
                        <p class="mb-2" style="font-size: 0.8rem">Download table with z-score (median expression) values
                            for <span
                                    class="gene-name"></span>
                            in various cell lines: <a id="fig5-down"
                                                      href="">tissue_and_cell_line_expression/<span
                                    class="gene-name"></span>_cell_line_zscore.csv</a></p>
                        <div class="modal fade" id="fig5-mod" tabindex="-1" aria-labelledby="fig5-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig5-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig5-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>

                    </div>
                </div>
                <h5 id="item-5" class="mt-4">Visualizing lncRNAs based on their gene expression similarity across tissues and cell
                    lines</h5>
                <p>We applied UMAP [6] to visualize lncRNA expression across 3,000 randomly selected samples (with
                    tissue type
                    and cell line labels) from Recount3[2]. Samples were first log2 transformed and quantile normalized
                    along
                    the gene axis, then UMAP was applied to the lncRNA expression data with samples as features. Each
                    data point
                    represents a single lncRNA (n=15,862). The black arrow is pointing to the location of <span
                            class="gene-name"></span>.</p>
                <div class="row">
                    <div class="col-sm">
                        <img id="fig6-img" src=""
                             class="img-fluid p-3 bg-light"
                             alt="">
                        <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig6-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 7. UMAP
                            was applied
                            to 3,000 randomly
                            selected samples (with tissue type labels) from Recount3.
                            Each data
                            point represents a lncRNA (n=15,862) and are colored by z-score (median expression) in <span
                                id="fig6-tissue"></span>.
                        </p>
                        <div class="modal fade" id="fig6-mod" tabindex="-1" aria-labelledby="fig6-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig6-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig6-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                    <div class="col-sm">
                        <img id="fig7-img" src=""
                             class="img-fluid p-3 bg-light"
                             alt="">
                        <p style="font-size: 0.8rem; font-style: italic;" class="mb-2 px-3 pb-2 bg-light">
                            <button type="button" class="btn btn-light btn-sm py-0" data-bs-toggle="modal"
                                    data-bs-target="#fig7-mod">
                                <i class="fas fa-expand-arrows-alt"></i>
                            </button>
                            Figure 8. UMAP
                            was applied
                            to 3,000 randomly
                            selected samples (with cell line labels) from Recount3. Each
                            data
                            point represents a lncRNA (n=15,862) and are colored by z-score (median expression) in <span
                                id="fig7-cell"></span>.
                        </p>
                        <div class="modal fade" id="fig7-mod" tabindex="-1" aria-labelledby="fig7-mod-title"
                             aria-hidden="true">
                            <div class="modal-dialog modal-lg">
                                <div class="modal-content">
                                    <div class="modal-header">
                                        <button type="button" class="btn-close" data-bs-dismiss="modal"
                                                aria-label="Close"></button>
                                    </div>
                                    <div class="modal-body">
                                        <img id="fig7-mod-img" src="" class="img-fluid" alt="">
                                    </div>
                                    <div class="modal-footer">
                                        <p class="modal-title" id="fig7-mod-title"></p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>

                <h5 id="item-6" class="mt-4">L1000 small molecules predicted to modulate <span class="gene-name"></span></h5>
                <p>~1.4 million Level 5 L1000 chemical perturbation gene expression signatures were downloaded from
                    SigCom LINCS
                    (https://maayanlab.cloud/sigcom-lincs) [7]. For each unique signature and lncRNA pair, a mean
                    Pearson
                    correlation coefficient was computed by taking the average Pearson coefficient between the lncRNA
                    and all genes
                    in the signature. All 15,862 lncRNAs were then ranked by mean Pearson correlation coefficient, and
                    the top 1,000
                    lncRNAs with the highest coefficients were retained for each signature. The top 500 lncRNA-L1000
                    signature
                    associations are reported here for <span class="gene-name"></span>, separated by direction. If <span
                            class="gene-name"></span> is highly
                    correlated with the
                    up-regulated genes for a specific small molecule, then this small molecule is predicted to
                    up-regulate
                    <span class="gene-name"></span>.</p>
                <h5 id="item-6-1" class="mt-4">L1000 small molecules predicted to up-regulate <span class="gene-name"></span></h5>
                <p>The prioritized small molecules below are predicted to specifically up-regulate <span
                        class="gene-name"></span>.
                </p>

                <div class="table-responsive-sm">
                    <p id="table3-blank" style="display: none; font-style: italic;">There are no small molecules
                        predicted to specifically up-regulate the expression of <span class="gene-name"></span>.</p>
                    <table id="table3" class="compact hover order-column row-border"
                           style="width:100%"></table>
                </div>

                <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a id="tab3-down"
                                                                                  href=""><i
                        class="fas fa-file-download"></i></a> Table 3. L1000 small molecules predicted to up-regulate
                    the lncRNA
                    <span class="gene-name"></span>.</p>

                <h5 id="item-6-2" class="mt-4">L1000 small molecules predicted to down-regulate <span class="gene-name"></span></h5>
                <p>The prioritized small molecules below are predicted to specifically down-regulate <span
                        class="gene-name"></span>.
                </p>

                <div class="table-responsive-sm">
                    <p id="table4-blank" style="display: none; font-style: italic;">There are no small molecules
                        predicted to specifically down-regulate the expression of <span class="gene-name"></span>.</p>
                    <table id="table4" class="compact hover order-column row-border"
                           style="width:100%"></table>
                </div>
                <p class="mb-2" style="font-size: 0.8rem; font-style: italic;"><a id="tab4-down"
                                                                                  href=""><i
                        class="fas fa-file-download"></i></a> Table 4. L1000 small molecules predicted to down-regulate
                    the lncRNA
                    <span class="gene-name"></span>.</p>


                <h5 id="item-7" class="mt-4">References</h5>
                <p class="mb-1" style="font-size: 0.8rem;">[1] Lorenz, R., Bernhart, S.H., HÃ¶ner zu Siederdissen, C. et
                    al. ViennaRNA Package 2.0. Algorithms Mol Biol 6, 26 (2011). https://doi.org/10.1186/1748-7188-6-26
                <p class="mb-1" style="font-size: 0.8rem;">[2] Wilks C, Zheng SC, Chen FY, Charles R, Solomon B, Ling
                    JP, Imada EL,
                    Zhang D, Joseph L, Leek JT: Recount3:
                    summaries and queries for large-scale RNA-seq expression and splicing. bioRxiv
                    2021:2021.2005.2021.445138.</p>
                <p class="mb-1" style="font-size: 0.8rem;">[3] Xie Z, Bailey A, Kuleshov MV, Clarke DJB, Evangelista JE,
                    Jenkins SL,
                    Lachmann A, Wojciechowicz ML,
                    Kropiwnicki E, Jagodnik KM: Gene Set Knowledge Discovery with Enrichr. Current Protocols 2021,
                    1(3):e90.</p>
                <p class="mb-1" style="font-size: 0.8rem;">[4] Chen EY, Tan CM, Kou Y, Duan Q, Wang Z, Meirelles GV,
                    Clark NR,
                    Maâ€™ayan A: Enrichr: interactive and
                    collaborative HTML5 gene list enrichment analysis tool. BMC Bioinformatics 2013, 14(1):128.</p>
                <p class="mb-1" style="font-size: 0.8rem;">[5] Kuleshov MV, Jones MR, Rouillard AD, Fernandez NF, Duan
                    Q, Wang Z,
                    Koplev S, Jenkins SL, Jagodnik KM,
                    Lachmann A: Enrichr: a comprehensive gene set enrichment analysis web server 2016 update. Nucleic
                    Acids Research
                    2016, 44(W1):W90-W97.</p>
                <p class="mb-1" style="font-size: 0.8rem;">[6] McInnes L, Healy J, Melville J: UMAP: Uniform manifold
                    approximation
                    and projection for dimension reduction.
                    arXiv preprint arXiv:180203426 2018.</p>
                <p class="mb-1" style="font-size: 0.8rem;">[7] Evangelista et al. SigCom LINCS: Data and Metadata Search
                    Engine for
                    Gene Expression Signatures. 2021. In
                    preparation.</p>


            </div>
        </div>
    </div>
    <!-- Results -->
</div>

{% endblock %}

{% block footer %}

<footer class="footer">

    <div class="container">
        <div class="row justify-content-center m-5">
            <div class="col-md-2 col-sm">
                <a href="https://icahn.mssm.edu/research/bioinformatics" target="_blank" rel="noreferrer">
                    <img
                            class="rounded"
                            src="{{ url_for('static', filename='icahn_cb_logo.png') }}"
                            style="height: 5rem"
                            alt="Icahn School of Medicine at Mount Sinai Center of Bioinformatics logo"
                    />
                </a>
            </div>
            <div class="col-md-2 col-sm">
                <a href="https://labs.icahn.mssm.edu/maayanlab/" target="_blank" rel="noreferrer">
                    <img
                            class="rounded"
                            src="{{ url_for('static', filename='maayanlab_logo.png') }}"
                            style="height: 5rem"
                            alt="Ma'ayan lab logo"
                    />
                </a>
            </div>
            <div class="col-md-2 col-sm ms-3">
                <div class="row">
                    <div class="col-12">
                    <!-- Twitted share button -->
                    <a href="https://twitter.com/intent/tweet"
                       class="twitter-share-button"
                       data-text="lncHUB2: Predicted biological functions for long non-coding RNAs based on co-expression"
                       data-url="https://{{ base_path }}/lncHUB2"
                       data-via="MaayanLab"
                       data-hashtags="lncRNA">Tweet</a>
                    </div>
                </div>
                <div class="row mb-1">
                    <div class="col-12">
                    <!-- LinkedIn share button -->
                    <script type="IN/Share" data-url="https://{{ base_path }}/lncHUB2"></script>
                    </div>
                </div>
                <!-- Facebook share button -->
                <div class="row">
                    <div class="col-12">
                    <div class="fb-share-button" data-href="https://maayanlab.cloud/lncHUB2" data-layout="button"
                         data-size="small"><a target="_blank"
                                              href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fmaayanlab.cloud%2FlncHUB2&amp;src=sdkpreparse"
                                              class="fb-xfbml-parse-ignore">Share</a></div>
                    </div>
                </div>
            </div>
                    <div class="col-md-2 col-sm">
            <a href="https://www.youtube.com/watch?v=yIsGmpNvTrU" target="_blank">
            <img id="youtube" style="border-radius: 5px; min-width: 175px" src="{{ url_for('static', filename='youtube_thumb.png') }}" class="img-fluid" alt="lncHUB2 Tutorial Video">
            </a>
        </div>
        </div>
    </div>
</footer>
{% endblock %}

{% block script %}
<script type="text/javascript" src="{{ url_for('static', filename='index.js') }}"></script>
{% endblock %}
